psi4/driver/procrouting/dft_funcs/dict_dh_funcs.py

```diff
@@ -0,0 +1,142 @@
+#
+# @BEGIN LICENSE
+#
+# Psi4: an open-source quantum chemistry software package
+#
+# Copyright (c) 2007-2017 The Psi4 Developers.
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This file is part of Psi4.
+#
+# Psi4 is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, version 3.
+#
+# Psi4 is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with Psi4; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+"""
+List of XC functionals
+"""
+
+
+b2plyp = {
+    "name": "B2PLYP",
+    "x_functionals": {"GGA_X_B88": {"alpha": 0.47}, "X_HF": {"alpha": 0.53}},
+    "c_functionals": {"GGA_C_LYP": {"alpha": 0.73}, "C_MP2": {"alpha": 0.27}},
```

Comment:
Not entirely sure I like `X_HF` and `C_MP2` as part of the "functionals" dictionary. Mostly this is because these need to be handled completely separately internally and usually I think of these as completely different components. I can easily be convinced, but its just something that struck me.

Reply:
We could change it to:
```
    ...
    "x_functionals": {"GGA_X_B88": {"alpha": 0.47}}, 
    "x_hf": {"alpha": 0.53},
    "c_functionals": {"GGA_C_LYP": {"alpha": 0.73}}, 
    "c_mp2": {"alpha": 0.27},
    ...
```
or even:
```
    "x_functionals": {"HYB_GGA_X_SOGGA11_X": {"use_libxc": True}}, # sets omega for GGA from libxc, alpha=1.0
    "x_hf": {"use_libxc": "HYB_GGA_X_SOGGGA11_X"}, # sets alpha, beta, omega for HF from libxc
    "c_functionals": {"GGA_C_SOGGA11_X": {}},
```

The question then remains what to do if one asks for a hybrid with `"use_libxc": True` and doesn't provide the `x_hf` key - do we add HF exchange as specified by libxc (even though the ALPHA actually applies for `x_hf` not for the GGAs in `x_functionals`), or do we just run it without HF exchange?

Reply:
I do like the change in format, but you are right we introduce some ambiguity.

I really wish that the SOGGA functionals (for example) just came as full XC functionals rather than the pieces. I dont quite understand why they deprecated the M05-2X macros and you now have to rebuild them on your own in LibXC 4.0+. This puts a lot of onus on the downstream users (us) to make sure this is correctly specified.

To make the parsing simpler I would do `{"HYB_GGA_X_SOGGA11_X": {"use_libxc": True}}` and then throw an error if `x_hf` has any data in it or if `x_functionals` has more than one functional in it. AFAIK there is no current functional that will violate this rule and we can handle it when the time arrises. 

I would say that the deprecation of XC functional macros (M05-2X and similar) puts us in a position of continuing to need expansive custom functions lists like this. Unless I am missing something?